src/components/PaymentInfoLine.js

```diff
@@ -0,0 +1,78 @@
+/* global Nimiq */
+/* global AddressInfo */
+/* global Timer */
+/* global TemplateTags */
+/* global NumberFormatting */
+
+class PaymentInfoLine { // eslint-disable-line no-unused-vars
+    /**
+     * @param {object}           paymentInfo
+     * @param {string}           paymentInfo.recipient
+     * @param {string|undefined} [paymentInfo.label]
+     * @param {URL|undefined}    [paymentInfo.imageUrl]
+     * @param {number}           paymentInfo.lunaAmount
+     * @param {number|undefined} [paymentInfo.fiatAmount]
+     * @param {string|undefined} [paymentInfo.fiatCurrency]
+     * @param {number|undefined} [paymentInfo.time]
+     * @param {number|undefined} [paymentInfo.expires]
+     * @param {HTMLElement}      [$el]
+     */
+    constructor(paymentInfo, $el) {
+        this.$el = PaymentInfoLine._createElement($el);
+
+        /** @type HTMLElement */
+        const $nimAmount = (this.$el.querySelector('.nim-amount'));
+        const nimAmount = NumberFormatting.formatNumber(Nimiq.Policy.lunasToCoins(paymentInfo.lunaAmount));
+        $nimAmount.textContent = `${nimAmount} NIM`;
+
+        /** @type HTMLElement */
+        const $fiatAmount = (this.$el.querySelector('.fiat-amount'));
+        if (paymentInfo.fiatAmount !== undefined && paymentInfo.fiatCurrency !== undefined) {
+            $fiatAmount.textContent = NumberFormatting.formatCurrency(paymentInfo.fiatAmount, paymentInfo.fiatCurrency);
+        } else {
+            $fiatAmount.remove();
+        }
+
+        const recipientInfo = new AddressInfo({
+            userFriendlyAddress: paymentInfo.recipient,
+            label: paymentInfo.label || null,
+            imageUrl: paymentInfo.imageUrl || null,
+            accountLabel: null,
+        });
+        recipientInfo.renderTo(/** @type HTMLElement */ (this.$el.querySelector('.recipient')));
+
+        /** @type HTMLElement */
+        const $timer = (this.$el.querySelector('.timer'));
+        if (paymentInfo.time !== undefined && paymentInfo.expires !== undefined) {
```

Comment:
I think both `time` and `expires` can also not be the number `0`, so it's safe to use truthy checks here:

```suggestion
        if (paymentInfo.time && paymentInfo.expires) {
```

Reply:
Changes this too.